shell/browser/event_emitter_mixin.h

```diff
@@ -39,6 +39,7 @@ class EventEmitterMixin {
                        v8::Local<v8::Object> custom_event,
                        Args&&... args) {
     v8::Isolate* isolate = v8::Isolate::GetCurrent();
+    v8::HandleScope scope(isolate);
```

Comment:
The `EmitCustomEvent` takes a `v8::Local` as parameter, I think we should add scope in the caller instead of here, otherwise the passed parameter is still leaked.

Reply:
@deepak1556 what do you think here? The former approach (from the caller) is a little weirder since we need to likely store the isolate as a member to prevent unnecessary loads, but i agree that we may wish to avoid leaking the passed parameter.

Reply:
Scoping `electron::api::CreateEventFromFlags` with an `EscapableHandleScope` should solve this instead, and other callers of `EmitCustomEvent` can do the same but right now there is only one usage, thoughts ?

Reply:
If there's a value returned from an escapable handle scope, then the calling function must have a handlescope anyway, so I don't see how that addresses the issue here?

Reply:
The value returned is passed directly to `EmitCustomEvent` , so the copied value lands in here rather than the function calling `EmitCustomEvent`, shouldn't that be safe ?

`EmitCustomEvent('..., CreateEventFromFlags(), ...');`

Reply:
But I am onboard with the singleton isolate holder, it will serve other use cases, this was just subtle that I was looking at other solutions.

Reply:
> The value returned is passed directly to `EmitCustomEvent` , so the copied value lands in here rather than the function calling `EmitCustomEvent`, shouldn't that be safe ?
> 
> `EmitCustomEvent('..., CreateEventFromFlags(), ...');`

No. Escapable handle scopes allow you to "push" a value into a handlescope that's higher up in the stack. If there's only one handlescope on the stack, "pushing" it up will fail.